WordPress/src/main/java/org/xmlrpc/android/ApiHelper.java

```diff
@@ -1086,21 +1099,41 @@ public static String getXMLRPCUrl(String urlString) throws SSLHandshakeException
      * @return content of the resource, or null if URL was invalid or resource could not be retrieved.
      */
     public static String getResponse(final String stringUrl) throws SSLHandshakeException {
+        return getResponse(stringUrl, 0);
+    }
+
+
+    public static String getResponse(final String stringUrl, int numberOfRedirects) throws SSLHandshakeException {
+        RequestFuture<String> future = RequestFuture.newFuture();
+        StringRequest request = new StringRequest(stringUrl, future, future);
+        WordPress.requestQueue.add(request);
         try {
-            URL url = new URL(stringUrl);
-            HttpURLConnection urlConnection = (HttpURLConnection) url.openConnection();
-            urlConnection.setInstanceFollowRedirects(true);
-            InputStream inputStream = urlConnection.getInputStream();
-            BufferedReader r = new BufferedReader(new InputStreamReader(inputStream));
-            StringBuilder sb = new StringBuilder();
-            String line;
-            while ((line = r.readLine()) != null) {
-                sb.append(line);
+            return future.get(XMLRPCClient.DEFAULT_SOCKET_TIMEOUT, TimeUnit.SECONDS);
+        } catch (InterruptedException e) {
+            AppLog.e(T.API, e);
+        } catch (ExecutionException e) {
+            if (e.getCause() != null && e.getCause() instanceof RedirectError) {
+                // Maximum 5 redirects or die
+                if (numberOfRedirects > 5) {
+                    AppLog.e(T.API, "Maximum of 5 redirects reached, aborting.", e);
+                    return null;
+                }
+                // Follow redirect
+                RedirectError re = (RedirectError) e.getCause();
+                if (re.networkResponse != null && re.networkResponse.headers != null
+                        && re.networkResponse.headers.containsKey("Location")) {
+                    String newURL = re.networkResponse.headers.get("Location");
+                    AppLog.i(T.API, "Follow redirect from " + stringUrl + " to " + newURL);
+                    // Abort redirect if old URL was HTTPS and not the new one
+                    if (URLUtil.isHttpsUrl(stringUrl) && !URLUtil.isHttpsUrl(newURL)) {
```

Comment:
The value of the Location header could be an absolute or a relative URL. This code seems to work fine on absolute URL redirect only.
See https://blackberry.trac.wordpress.org/browser/trunk/src/com/wordpress/xmlrpc/HTTPPostConn.java#L82